pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/reader/H3IndexResolution.java

```diff
@@ -91,4 +94,11 @@ public boolean equals(Object o) {
   public int hashCode() {
     return Objects.hash(_resolutions);
   }
+
+  public static class ToIntListConverted extends StdConverter<H3IndexResolution, List<Integer>> {
```

Comment:
was wondering if we can do 
- serialize just the short format and not to List<Integer> serialization?
- still keep the JSON creator but also annotate the short constructor as JSON create?

Reply:
With the custom deserializer, `@JsonCreator` is not needed (and in fact it may be confusing). I decided to serialize as list, but we can change that to deserialize as short if you think that is better. I guess neither short or list are human readable and short is always going to be smaller.

Reply:
can we check the serialized size for comparison?

Reply:
Before this PR, the actual value that was serialized was a string like `"5,6,13"`, whose size codified as a JSON is 8 characters. The new codification is the same (`[5,6,13]`). The larger value that can be represented like that is `[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15]`, which is 35 characters (and TBH I don't know if it is a valid resolution).

On the other hand, the numeric codification is a short, whose larger utf8 value would be `-32768`, which is 6 characters.

Sometimes the array codification will be better. For example, `[14]` requires 4 chars, while it would be represented as `16384` in the short codification. But there the values where this happens is very small.

Reply:
do we need the list format? as both are not human readable eitherway. keeping it as short is good IMO. 
as long as we can construct this from list or short we should be fine right?

Reply:
cool. good to see this improvement